designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala

```diff
@@ -37,6 +40,14 @@ class DefinitionResources(modelDataProvider: ProcessingTypeDataProvider[ModelDat
           modelDataProvider.mapValues(_.processDefinition.services.mapValuesNow(UIProcessObjectsFactory.createUIObjectDefinition(_, processCategoryService))).all
         }
       }
+    } ~ path("processDefinitionData" / "categoriesWithProcessingType") {
+      get {
+        complete {
+            processCategoryService.getUserCategories(user)
```

Comment:
Hmm, maybe better way is to create one method for userCategoriesType?

Reply:
No problem'o - I've added `getUserCategoriesWithType` method and it helped me to drop `Option` from return type without any handlers. :+1: